packages/shared/src/collections/dto/update-collection-request.ts

```diff
@@ -1,17 +1,50 @@
-import {Config} from "../config";
+import { Static, Type } from "@sinclair/typebox";
+import {Config, ConfigType} from "../config";
 
-export interface UpdateCollectionRequest {
-    displayName: string;
-    configs: Config[];
-}
+export const ConfigValidation = Type.Union([
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.CHECKBOX),
+        value: Type.Boolean(),
+    }),
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.SHORT_TEXT),
+        value: Type.String(),
+    }),
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.LONG_TEXT),
+        value: Type.String(),
+    }),
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.NUMBER),
+        value: Type.Number(),
+    }),
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.DICTIONARY),
+        value: Type.Object({}),
+    }),
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.CLOUD_OAUTH2),
+        settings: Type.Object({}),
+        value: Type.Object({}),
+    }),
+    Type.Object({
+        key: Type.String(),
+        type: Type.Literal(ConfigType.OAUTH2),
+        settings: Type.Object({}),
+        value: Type.Object({}),
+    })
+    
```

Comment:
This comment is irrelevant to this PR, but do we need to add a `ConfigType` for `CUSTOM_AUTH`?

Reply:
Nope I will remove Authentication configs once I merged app-auth branch, but i couldn't make this branch compiles without them